hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1495,12 +1495,25 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, fmt.Errorf("failed to reconcile control plane operator: %w", err)
 	}
 
+	configOperatorImage, releaseVersion, err := func() (string, string, error) {
+		releaseInfo, err := r.ReleaseProvider.Lookup(ctx, hcluster.Spec.Release.Image, pullSecretBytes)
```

Comment:
Performance/responsiveness impact here.  This `Lookup` is never hit in modern hypershift because the CPO manages Autoscaler, MachineApprover, and IgnitionServer.  If not cached yet, this is a synchronous pull of a release image in the reconcile loop.